tests/fixtures/optimizer/qualify_columns.sql

```diff
@@ -72,6 +72,9 @@ SELECT x.a AS a, x.b AS b FROM x AS x ORDER BY a;
 SELECT a FROM x ORDER BY b;
 SELECT x.a AS a FROM x AS x ORDER BY x.b;
 
+SELECT SUM(a) AS a FROM x ORDER BY SUM(a);
```

Comment:
i assume you tested this case in the various engines and this is the behavior?

Reply:
this mimics the mysql behavior for whether it resolves to an alias or a column. Starrocks always resolves to an alias if there is one which leads to unexpected behavior in situations like this - the ordering was random when I tested this scenario.